### PR TITLE
[v10.1.x] Docs: Remove old plugin docs, fix URLs to go to new dev portal (#75325)

### PR DESCRIPTION
**Pull request by jdbaldry**
* Remove old plugin docs, add redirects to new dev portal

* Restore plugin.schema.json file

* Update docs/sources/administration/plugin-management/index.md

* More updates

* Cleanup links

* Update docs/sources/alerting/fundamentals/evaluate-grafana-alerts.md

* fix codeowners

* Change reference links to inline links to avoid 404s

* Update docs/sources/datasources/_index.md

* Update docs/sources/datasources/mysql/_index.md

* Update docs/sources/datasources/postgres/_index.md

* Update docs/sources/fundamentals/timeseries-dimensions/index.md

* Update docs/sources/fundamentals/dashboards-overview/index.md

* Update docs/sources/fundamentals/dashboards-overview/index.md

* Update docs/sources/panels-visualizations/configure-standard-options/index.md

* Update docs/sources/panels-visualizations/query-transform-data/transform-data/index.md

* Remove grafana.com from 2 URLs

* Fix 1 more reference link

* Prettier fix

---------

(cherry picked from commit 0e2b741fc33ae5382fb27b4659f2b3ecd13f2afe)

### CODE REVIEW

File: .changelog-archive/CHANGELOG.7.md [hunk 1]
```diff
@@ -1294,7 +1294,7 @@ This option to group query variable values into groups by tags has been an exper
 **Deprecation warnings**
 
 - Scripted dashboards is now deprecated. The feature is not removed but will be in a future release. We hope to address the underlying requirement of dynamic dashboards in a different way. [#24059](https://github.com/grafana/grafana/issues/24059)
-- The unofficial first version of backend plugins together with usage of [grafana/grafana-plugin-model](https://github.com/grafana/grafana-plugin-model) is now deprecated and support for that will be removed in a future release. Please refer to [backend plugins documentation](https://grafana.com/docs/grafana/latest/developers/plugins/backend/) for information about the new officially supported backend plugins.
+- The unofficial first version of backend plugins together with usage of [grafana/grafana-plugin-model](https://github.com/grafana/grafana-plugin-model) is now deprecated and support for that will be removed in a future release. Please refer to [backend plugins documentation](/developers/plugin-tools) for information about the new officially supported backend plugins.
```
Comment (sympatheticmoose) [inline .changelog-archive/CHANGELOG.7.md:1297]:
```suggestion
- The unofficial first version of backend plugins together with usage of [grafana/grafana-plugin-model](https://github.com/grafana/grafana-plugin-model) is now deprecated and support for that will be removed in a future release. Please refer to [backend plugins documentation](/developers/plugin-tools/introduction/backend-plugins) for information about the new officially supported backend plugins.
```

**Comment by jdbaldry** [comment]
This is a backport so if I incorporate these changes, we will probably need someone to make them also in `main`. 

cc. @josmperez

File: contribute/breaking-changes-guide.md [hunk 2]
```diff
@@ -104,7 +104,7 @@ myOldFunction(name: string) {
 2. Add info in the comment about **when it is going to be removed**
 3. Add info in the comment about **what should be used instead**
 4. In case it's a function or a method, use `deprecationWarning(<file name>, <old name>, <new name>)` to raise attention during runtime as well
-5. Update the [migration guide](../docs/sources/developers/plugins/migration-guide.md) with your instructions
+5. Update the [migration guide](/developers/plugin-tools/migration-guides/) with your instructions
```
Comment (sympatheticmoose) [inline contribute/breaking-changes-guide.md:107]:
```suggestion
5. Update the [migration guidance](/developers/plugin-tools/migration-guides/) for plugin developers with your instructions
```

File: docs/sources/administration/data-source-management/index.md [hunk 3]
```diff
@@ -195,4 +195,4 @@ For more documentation on a specific data source plugin's features, including it
 
 ### Create a data source plugin
 
-To build your own data source plugin, refer to the ["Build a data source plugin"](/tutorials/build-a-data-source-plugin/) tutorial and our documentation about [building a plugin](/developers/plugins/).
+To build your own data source plugin, refer to the ["Build a data source plugin"](/developers/plugin-tools/tutorials/build-a-data-source-plugin) tutorial and our documentation about [building a plugin](/developers/plugins-tools).
```
Comment (sympatheticmoose) [inline docs/sources/administration/data-source-management/index.md:198]:
```suggestion
To build your own data source plugin, refer to the ["Build a data source plugin"](/developers/plugin-tools/tutorials/build-a-data-source-plugin) tutorial and our documentation about [building plugins](/developers/plugins-tools).
```

File: docs/sources/datasources/_index.md [hunk 4]
```diff
@@ -29,7 +29,7 @@ After you add and configure a data source, you can use it as an input for many o
 This documentation describes how to manage data sources in general,
 and how to configure or query the built-in data sources.
 For other data sources, refer to the list of [datasource plugins](/grafana/plugins/).
-To develop a custom plugin, refer to [Plugin tools](/developers/plugin-tools).
+To develop a custom plugin, refer to [Build a plugin](/developers/plugin-tools)
```
Comment (sympatheticmoose) [inline docs/sources/datasources/_index.md:32]:
```suggestion
To create a custom plugin, refer to our [plugin development documentation](/developers/plugin-tools)
```

File: docs/sources/datasources/mysql/_index.md [hunk 5]
```diff
@@ -297,7 +297,7 @@ The examples in this section query the following table:
 
 If the `Format as` query option is set to `Time Series` then the query must have a column named time that returns either a SQL datetime or any numeric datatype representing Unix epoch in seconds. In addition, result sets of time series queries must be sorted by time for panels to properly visualize the result.
 
-A time series query result is returned in a [wide data frame format](/developers/plugin-tools/introduction/data-frames#wide-format). Any column except time or of type string transforms into value fields in the data frame query result. Any string column transforms into field labels in the data frame query result.
+A time series query result is returned in a [wide data frame format](https://grafana.com/developers/plugin-tools/introduction/data-frames#wide-format). Any column except time or of type string transforms into value fields in the data frame query result. Any string column transforms into field labels in the data frame query result.
```
Comment (sympatheticmoose) [inline docs/sources/datasources/mysql/_index.md:300]: @jdbaldry out of curiosity - why is this one an absolute link?

**Comment by sympatheticmoose** [comment]
@jdbaldry good point... I'll continue to raise as suggestions, but don't feel like you have to action. If @josmperez agrees with the suggestions maybe we open a new PR on main then backport that one? What would be easiest/cleanest?

File: docs/sources/developers/_index.md [hunk 6]
```diff
@@ -11,12 +11,18 @@ weight: 190
 
 # Developers
 
-This section includes the following Grafana developer documentation:
+Go to the [Grafana developer portal](/developers) to access the following documentation:
+
+- [Grafana plugin development](/developers/plugin-tools)
+- [Grafana design system](https://developers.grafana.com)
```
Comment (sympatheticmoose) [inline docs/sources/developers/_index.md:17]:
```suggestion
- [Grafana UI component library](https://developers.grafana.com)
```

File: docs/sources/fundamentals/timeseries-dimensions/index.md [hunk 7]
```diff
@@ -104,8 +104,6 @@ In the case of SQL-like data sources, more than one numeric column can be select
 Additional technical information on tabular time series formats and how dimensions are extracted can be found in [the developer documentation on data frames as time series](/developers/plugin-tools/introduction/data-frames#data-frames-as-time-series).
 
 {{% docs/reference %}}
```
Comment (sympatheticmoose) [inline docs/sources/fundamentals/timeseries-dimensions/index.md:106]: This doesn't seem related to plugin docs? Commenting just in case

File: pkg/plugins/plugindef/plugindef_types_gen.go
Comment (sympatheticmoose) [inline pkg/plugins/plugindef/plugindef_types_gen.go:1]: is this safe to remove @andresmgot ?

**Review by chri2547 — APPROVED** [review]
The docs that impact my area I'm fine with. Thank you!

File: docs/sources/panels-visualizations/query-transform-data/transform-data/index.md [hunk 8]
```diff
@@ -974,4 +974,27 @@ Here is the result after adding a Limit transformation with a value of '3':
 This transformation is available in Grafana 9.5+ as an opt-in beta feature. Modify Grafana [configuration file]({{< relref "../../../setup-grafana/configure-grafana/#configuration-file-location" >}}) to enable the `timeSeriesTable` [feature toggle]({{< relref "../../../setup-grafana/configure-grafana/#feature_toggles" >}}) to use it.
 {{% /admonition %}}
 
-Use this transformation to convert time series result into a table, converting time series data frame into a "Trend" field. "Trend" field can then be rendered using [sparkline cell type]({{< relref "../../visualizations/table/#sparkline" >}}), producing an inline sparkline for each table row. If there are multiple time series queries, each will result in a separate table data frame. These can be joined using join or merge transforms to produce a single table with multiple sparklines per row.
+Use this transformation to format the output of a time field. Output can be formatted using (Moment.js format strings)[https://momentjs.com/docs/#/displaying/]. For instance, if you would like to display only the year of a time field the format string `YYYY` can be used to show the calendar year (e.g. 1999, 2012, etc.).
```
Comment (jdbaldry) [inline docs/sources/panels-visualizations/query-transform-data/transform-data/index.md:977]:
```suggestion
Use this transformation to format the output of a time field. Output can be formatted using [Moment.js format strings](https://momentjs.com/docs/#/displaying/. For instance, if you would like to display only the year of a time field the format string `YYYY` can be used to show the calendar year (e.g. 1999, 2012, etc.).
```

File: docs/sources/panels-visualizations/query-transform-data/transform-data/index.md [hunk 9]
```diff
@@ -974,4 +974,27 @@ Here is the result after adding a Limit transformation with a value of '3':
...
+
+[configuration file]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/setup-grafana/configure-grafana#configuration-file-location"
+[configuration file]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/setup-grafana/configure-grafana#configuration-file-location"
+
+[Time series panel]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/panels-visualizations/visualizations/time-series"
+[Time series panel]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/panels-visualizations/visualizations/time-series"
+
+[feature toggle]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/setup-grafana/configure-grafana#feature_toggles"
+[feature toggle]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/setup-grafana/configure-grafana#feature_toggles"
+{{% /docs/reference %}}
```
Comment (jdbaldry) [inline docs/sources/panels-visualizations/query-transform-data/transform-data/index.md:1000]:
```suggestion
```

File: docs/sources/panels-visualizations/configure-standard-options/index.md [hunk 10]
```diff
@@ -146,3 +146,8 @@ Select one of the following palettes:
 ### No value
 
 Enter what Grafana should display if the field value is empty or null. The default value is a hyphen (-).
+
+{{% docs/reference %}}
+[variables]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/dashboards/variables"
+[variables]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/dashboards/variables"
+{{% /docs/reference %}}
```
Comment (jdbaldry) [inline docs/sources/panels-visualizations/configure-standard-options/index.md:153]:
```suggestion
```

File: docs/sources/fundamentals/timeseries-dimensions/index.md (hunk 7 above)
Comment (jdbaldry) [reply docs/sources/fundamentals/timeseries-dimensions/index.md:106]: Awesome catch! This does indeed look wrong.

File: docs/sources/datasources/mysql/_index.md (hunk 5 above)
Comment (jdbaldry) [reply docs/sources/datasources/mysql/_index.md:300]: That's an interesting question. Historically, the use of partial URLs was encouraged to be sympathetic to the staging site which has a different hostname. Now this is no longer necessary but there is the linter in place for some of these docs still insists on the use of partial URLs for now. This doc must be one that is not yet linted.

File: docs/sources/alerting/fundamentals/evaluate-grafana-alerts.md [hunk 11]
```diff
@@ -30,7 +30,7 @@ Grafana managed alerts query the following backend data sources that have alerti
 
 - built-in data sources or those developed and maintained by Grafana: `Graphite`, `Prometheus`, `Loki`, `InfluxDB`, `Elasticsearch`,
   `Google Cloud Monitoring`, `Cloudwatch`, `Azure Monitor`, `MySQL`, `PostgreSQL`, `MSSQL`, `OpenTSDB`, `Oracle`, and `Azure Monitor`
-- community developed backend data sources with alerting enabled (`backend` and `alerting` properties are set in the [plugin.json](/developers/plugin-tools/reference-plugin-json))
+- community developed backend data sources with alerting enabled (`backend` and `alerting` properties are set in the [plugin.json](/developers/plugin-tools/reference-plugin-json)
```
Comment (jdbaldry) [inline docs/sources/alerting/fundamentals/evaluate-grafana-alerts.md:33]:
```suggestion
- community developed backend data sources with alerting enabled (`backend` and `alerting` properties are set in the [plugin.json](/developers/plugin-tools/reference-plugin-json))
```

**Comment by imatwawana** [comment]
I've tried building just `grafana` locally and `website` and `grafana` and I still can't get the links I'm checking to work. Not sure if I'm building it incorrectly or if these should work at all. If there's something else I need to do to test this, let me know.

**Comment by jdbaldry** [comment]
> I've tried building just `grafana` locally and `website` and `grafana` and I still can't get the links I'm checking to work. Not sure if I'm building it incorrectly or if these should work at all. If there's something else I need to do to test this, let me know.

Unfortunately `/developers/` is not part of the Hugo site and is not available during the local build currently.

File: docs/sources/panels-visualizations/query-transform-data/transform-data/index.md [hunk 12]
```diff
@@ -974,4 +974,4 @@ Here is the result after adding a Limit transformation with a value of '3':
 This transformation is available in Grafana 9.5+ as an opt-in beta feature. Modify Grafana [configuration file]({{< relref "../../../setup-grafana/configure-grafana/#configuration-file-location" >}}) to enable the `timeSeriesTable` [feature toggle]({{< relref "../../../setup-grafana/configure-grafana/#feature_toggles" >}}) to use it.
 {{% /admonition %}}
 
-Use this transformation to convert time series result into a table, converting time series data frame into a "Trend" field. "Trend" field can then be rendered using [sparkline cell type]({{< relref "../../visualizations/table/#sparkline" >}}), producing an inline sparkline for each table row. If there are multiple time series queries, each will result in a separate table data frame. These can be joined using join or merge transforms to produce a single table with multiple sparklines per row.
+Use this transformation to format the output of a time field. Output can be formatted using [Moment.js format strings](https://momentjs.com/docs/#/displaying/. For instance, if you would like to display only the year of a time field the format string `YYYY` can be used to show the calendar year (e.g. 1999, 2012, etc.).
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/query-transform-data/transform-data/index.md:977]: This is the content from a new transformation that's in the next docs but not in latest. 
```suggestion
```

**Review by imatwawana — COMMENTED** [review]
Found one problem with the panels & visualizations content.

**Review by josmperez — APPROVED** [review]
LGTM.

**Comment by josmperez** [comment]
> If @josmperez agrees with the suggestions maybe we open a new PR on main then backport that one? What would be easiest/cleanest?

These suggestions made sense and I added them to the backport for v10.0.x and opened [a PR](https://github.com/grafana/grafana/pull/75839) for making these fixes to main.
